sunkit_instruments/suvi/suvi.py

```diff
@@ -0,0 +1,663 @@
+    """
+    Despike SUVI L1b data and return either a despiked
+    `~numpy.ndarray` or a despiked `~sunpy.map.Map`.
+
+    .. note::
+        If the the_input is a string: the type of file is determined
+        by pattern matching in the filenames, e.g. "-L1b-Fe171"
+        for a 171 L1b file. If this pattern is not found in the
+        filename, the file will not be recognized.
+
+    .. note::
+        The despiking relies on the presence of the data quality
+        flags (DQF) in the first extension of a SUVI L1b FITS file.
+        Early in the mission, the DQF extension was not present
+        yet, so the despiking cannot be done with this function
+        for those early files.
+
+    Parameters
+    ----------
+    the_input: `str` or `tuple` of 2-D `~numpy.ndarray` containing (data, DQF)
+        Filename or tuple of (data, DQF) to despike.
+
+    filter_width: `int`, optional. Default: 7.
+        The filter width for the Gaussian filter. If NaNs are still
+        present in the despiked image, try increasing this value.
+
+    return_map: `bool`, optional. Default: False.
+        If True, returns a `~sunpy.map.Map` instead of a `~numpy.ndarray`.
+        Ignored if the_input is not a filename string.
+
+    Returns
+    -------
+    despiked_image: `~numpy.ndarray`
+        The despiked L1b image as a numpy array.
+
+    map: `~sunpy.map.Map` if return_map=True.
+        The despiked L1b image as a `~sunpy.map.Map`.
+    """
+
+    if isinstance(the_input, tuple):
+        image = the_input[0]
+        dqf_mask = the_input[1]
+    elif isinstance(the_input, str):
+        L1b_matches = ['-L1b-Fe093', '-L1b-Fe131', '-L1b-Fe171', '-L1b-Fe195', '-L1b-Fe284', '-L1b-He303']
+        if any(fn in os.path.basename(the_input) for fn in L1b_matches):
+            header, image, dqf_mask = sunkit_instruments.suvi.read_suvi(the_input, return_DQF=True)
+        else:
+            raise ValueError("File "+the_input+" does not look like a SUVI L1b file.")
+    else:
+        raise TypeError("Input must be string or tuple of 2-D arrays.")
+
+    image_with_nans = numpy.copy(image)
+    image_with_nans[numpy.where(dqf_mask == 4)] = numpy.nan
+    indices = numpy.where(numpy.isnan(image_with_nans))
+    image_gaussian_filtered = gaussian_filter(image, filter_width)
+    despiked_image = numpy.copy(image_with_nans)
+    despiked_image[indices] = image_gaussian_filtered[indices]
+
+    if return_map and isinstance(the_input, str):
+        return sunpy.map.Map(despiked_image, header)
+    else:
+        return despiked_image
+
+
+# helper function: parser for the SUVI websites using BeautifulSoup
+def _list_url_directory(url, ext=''):
+    page = requests.get(url).text
+    soup = BeautifulSoup(page, 'html.parser')
+    return [url + node.get('href') for node in soup.find_all('a') if node.get('href').endswith(ext)]
+
+
+def download_data_from_NOAA(date_time, spacecraft=16, wavelength=171, outdir=None,
```

Comment:
I forgot about this but we already do have a client: https://github.com/sunpy/sunpy/blob/main/sunpy/net/dataretriever/sources/goes.py#L174

It should offer the same range of files as this function.

Reply:
Thanks, I was unaware of that. I will just remove my function then.

Reply:
You might want to double check, if not, we can update the client in sunpy.